packages/near-membrane-base/src/environment.ts

```diff
@@ -346,4 +354,10 @@ export class VirtualEnvironment {
             this.redCallableSetPrototypeOf(foreignTargetPointer, transferableProto);
         }
     }
+
+    trackAsFastTarget(target: ProxyTarget) {
+        if ((typeof target === 'object' && target !== null) || typeof target === 'function') {
+            this.redCallableTrackAsFastTarget(this.blueGetTransferableValue(target) as Pointer);
```

Comment:
☝️ Because with we have easy access to the `foreignTargetPointer` we track fast targets through their foreign target pointers. So to do that from the exposed `trackAsFastTarget` we have a bit of juggling to pass the `target` as a foreign target pointer. This saves on checks everywhere else in the code.